nncf/torch/model_creation.py

```diff
@@ -326,6 +326,8 @@ def wrap_model(
     :param trace_parameters: Whether to trace model parameters. Default is False.
     :return: A model wrapped by NNCFNetwork.
     """
+    if not isinstance(model, torch.nn.Module):
+        raise ValueError("Invalid model. Only PyTorch models inherited from torch.nn.Module are supported")
```

Comment:
```suggestion
        raise TypeError(f"The type if provided model {type(model)} is not compatible. Only models that inherit from torch.nn.Module are supported.)
```
What think about it?
ValueError is about incorrect value of argument, like -1 when expected value >0. TypeError is more correct in this case.